lib/Dialect/FIRRTL/Export/FIREmitter.cpp

```diff
@@ -287,6 +287,15 @@ struct Emitter {
   DenseMap<Value, StringRef> valueNames;
   StringSet<> valueNamesStorage;
 
+  /// Legalize names for emission.  Convert names which begin with a number to
+  /// be escaped using backticks.
+  StringAttr legalize(StringAttr attr) {
+    StringRef str = attr.getValue();
+    if (str.empty() || !isdigit(str.front()))
+      return attr;
+    return StringAttr::get(attr.getContext(), "`" + Twine(attr) + "`");
```

Comment:
I was going to suggest letting PrettyPrinter's string storage handle this for you (the thing `PPExtString` opts out of) instead of unique'ing into the context, but probably this is something already done as part of IR printing/etc., and also sufficiently rare to not matter.